### ISSUE TITLE
2.16

### ISSUE THREAD

**Issue by karato** [yegor256/netbout#683]
Please, publish a new release **2.16** (or higher). Don't forget to update us here regularly (at least once a week), as explained in [this article](http://www.yegor256.com/2015/05/11/software-architect-responsibilities.html).

**Comment by karato** [comment]
@dmzaytsev this is your task

**Comment by karato** [comment]
@dmzaytsev there is no milestone yet, so I set it to `3.1`

**Comment by karato** [comment]
@dmzaytsev there are some issues in the project now:

 * some tasks are stuck: #567 (107 days), #614 (24 days), #612 (24 days), #606 (24 days), #603 (24 days), #602 (24 days); please do something about it. There are basically three options: 1) mark some of them as "postponed" (see [par.26](http://www.teamed.io/policy.html#26)), 2) re-assign them (see [par.29](http://www.teamed.io/policy.html#29)), or 3) try to help their performers to resolve them.

 * some issues were reported some time ago, but still not dispatched: #658, #656, #644, #634, #626, #622. Please, do something about them. Either register as bugs or ask their reporters to close them.

**Comment by dmzaytsev** [comment]
@karato 
>some tasks are stuck:

* #567 should be closed soon
* #614 seems need to be reassigned to another developer
* #612 we have the impediment  for this task
* #606 I removed "postponed" tag 4 days ago, task assigned to the developer
* #603 shoud be closed soon
* #602 I removed "postponed" tag 4 days ago, task assigned to the developer


>some issues were reported some time ago, but still not dispatched: 

* #658 asked the reporter to close it.
* #656 I will update the javadoc 
* #644 hard to reproduce the bug, I wouldn't like dispatch this ticket right now
* #634 registered as a bug
* #626 registered as a bug
* #622 asked to be assigned to @yegor256

**Comment by dmzaytsev** [comment]
@yegor256 please see the project report

Scope status:

* RESTful API [95%]
* search functionality [40%]
* email enhancement [55%]
* UI enhancement[10%]
* current bugs fix [20%]

Issues:
I need your advise in these tickets
* https://github.com/yegor256/netbout/issues/697#issuecomment-111784729
* https://github.com/yegor256/netbout/issues/602#issuecomment-110858137
* https://github.com/yegor256/netbout/issues/622#issuecomment-107317733
* What we shall to do with tickets like #644 ? we are definitely know a bug exists, but can't reproduced it. A first question from developer will be "how I can reproduce it?"


Risks:

* 3.0 release make take more time. It's a campus project [7/6]

**Comment by karato** [comment]
@dmzaytsev There are a few concerns:

 * some tasks are stuck: #567 (110 days), #614 (28 days), #612 (28 days), #606 (28 days), #603 (27 days), #602 (27 days), #641 (21 days), #637 (21 days); please do something about it. There are basically three options: 1) mark some of them as "postponed" (see [par.26](http://www.teamed.io/policy.html#26)), 2) re-assign them (see [par.29](http://www.teamed.io/policy.html#29)), or 3) try to help their performers to resolve them.

 * these issues are still not dispatched anyhow: #688, #677, #658, #656, #644, #622. Either register as bugs or ask their reporters to close.

**Comment by yegor256** [comment]
@dmzaytsev it's bad to have "stuck" tasks. You should either postpone them or re-assign. But we can't keep open and assigned for 20+ days. It's bad for project manageability

**Comment by dmzaytsev** [comment]
@yegor256 if an  issue marked as postponed  it means the issue never will be resolved, isn't it.
I think waiting is the decision, if a task has impediment.
When impediment will be resolved, the PM will inform us.

**Comment by yegor256** [comment]
@dmzaytsev yes, waiting is fine, but for example #614 is not waiting for anything, etc. We should try to avoid such situations. It's not always obvious how to avoid them, but you should keep an eye on them. Also, keep in mind this: http://www.teamed.io/policy.html#31 If some task is stuck because it's too difficult, we can boost it and increase its budget a few times. This will motivate its developer to hurry up.

**Comment by dmzaytsev** [comment]
@yegor256 
>some tasks are stuck: 

#567 (110 days) closed
#614 (28 days), assigned 15 days ago, should be reassigned today. I will think what  I can be done with this task.
#612 (28 days), has the impediment
#606 (28 days), assigned 9 days ago, I reminded the developer 
#603 (27 days), I asked the reporter to stretch the point and close the task
#602 (27 days), assigned 9 days ago, I reminded the developer 
#641 (21 days), has the impediment
#637 (21 days); has the impediment


>not dispatched anyhow: 

#688 closed
#677 I asked mark it as postponed 6 days ago...
#658 I asked to close it.
#656 a cosmetic issue
#644 unable to reproduce, I wouldn't like dispatch this ticket,  needs your advise
#622. needs your advise

@yegor256 can we make a separate list for stucked tasks and tasks which has an impediment?

**Comment by dmzaytsev** [comment]
@yegor256 seems there is some issues with a project automation ....
e.g. these tasks  #564 #706 #677 has not been handled by PM a few days

**Comment by dmzaytsev** [comment]
@yegor256 can we publish 2.16?

**Comment by karato** [comment]
@dmzaytsev There are a few problems in the project now:

 * some tasks are stuck: #614 (31 days), #606 (31 days), #602 (31 days), #632 (25 days), #628 (25 days); please do something about it. There are basically three options: 1) mark some of them as "postponed" (see [par.26](http://www.teamed.io/policy.html#26)), 2) re-assign them (see [par.29](http://www.teamed.io/policy.html#29)), or 3) try to help their performers to resolve them.

 * some issues were reported some time ago, but still not dispatched: #697, #677, #658, #656, #644, #622. Please, do something about them. Either register as bugs or ask their reporters to close them.

**Comment by karato** [comment]
@dmzaytsev There are a few concerns:

 * some tasks are stuck: #614 (34 days), #606 (34 days), #602 (34 days), #628 (28 days); please resolve this. You have three options: 1) mark some of them as "postponed", 2) re-assign them, or 3) help performers to resolve them.

 * we have too many (7) urgent tasks, it's not healthy. Please, de-prioritize some of them.

 * these issues are still not dispatched anyhow: #708, #706, #677, #658, #656, #644, #622. Either register as bugs or ask their reporters to close.

**Comment by dmzaytsev** [comment]
@yegor256 how many urgent task is acceptable for the project ?
We have some urgent tasks  not assigned yet or will become unassigned soon

**Comment by dmzaytsev** [comment]
@yegor256 @karato 

** stuck tasks **
#614 (34 days) not dispached yet, 
#606 (34 days) under the code review now,  I hope will be resolved soon
#602 (34 days) the performer not respond I think will be reassigned in two days
#628 (28 days) I marked it as urgent 3 days ago should be dispatched soon

**not dispatched anyhow:**
#708 сlosed, 
#706 marked as bug and should be on hold 6 days ago
#677 marked as bug and postponed 10 days ago!
#658 asked the reporter to close 
#656 asked the reporter to close,  a cosmetic issue
#644 unable to reproduce, I wouldn't like dispatch this ticket, needs your advise
#622 needs your advise

**Comment by yegor256** [comment]
@dmzaytsev five urgent tasks is a reasonable maximum. If we have more, we're losing the whole point of marking them "urgent"

**Comment by karato** [comment]
@dmzaytsev there are some issues in the project now:

 * some tasks are stuck: #614 (37 days), #602 (37 days), #628 (31 days); please do something about it. There are basically three options: 1) mark some of them as "postponed" (see [par.26](http://www.teamed.io/policy.html#26)), 2) re-assign them (see [par.29](http://www.teamed.io/policy.html#29)), or 3) try to help their performers to resolve them.

 * these issues are still not dispatched anyhow: #677, #658, #656, #644, #622. Either register as bugs or ask their reporters to close.

**Comment by karato** [comment]
@dmzaytsev there are some issues in the project now:

 * some tasks are stuck: #614 (41 days), #602 (41 days), #641 (35 days); please do something about it. There are basically three options: 1) mark some of them as "postponed" (see [par.26](http://www.teamed.io/policy.html#26)), 2) re-assign them (see [par.29](http://www.teamed.io/policy.html#29)), or 3) try to help their performers to resolve them.

 * these issues are still not dispatched anyhow: #656, #644, #622. Either register as bugs or ask their reporters to close.

**Comment by karato** [comment]
@dmzaytsev please, pay attention to this:

 * some tasks are stuck: #614 (47 days), #602 (47 days), #641 (41 days), #692 (24 days), #698 (22 days), #634 (22 days), #626 (22 days); please resolve this. You have three options: 1) mark some of them as "postponed", 2) re-assign them, or 3) help performers to resolve them.

 * some issues were reported some time ago, but still not dispatched: #656, #644, #622. Please, do something about them. Either register as bugs or ask their reporters to close them.

 * there are some milestones with missed dates: [3.0](https://api.github.com/repos/yegor256/netbout/milestones/1). Please, do something about it.

**Comment by antonini** [comment]
@karato Issue #614 is waiting for you action and this depends on #728.

**Comment by karato** [comment]
@dmzaytsev please, pay attention to this:

 * some tasks are stuck: #614 (50 days), #602 (50 days), #641 (44 days), #692 (27 days), #698 (25 days), #634 (25 days), #626 (25 days), #704 (22 days); please do something about it. There are basically three options: 1) mark some of them as "postponed" (see [par.26](http://www.teamed.io/policy.html#26)), 2) re-assign them (see [par.29](http://www.teamed.io/policy.html#29)), or 3) try to help their performers to resolve them.

 * some issues were reported some time ago, but still not dispatched: #656, #644, #622. Please, do something about them. Either register as bugs or ask their reporters to close them.

 * some milestones are already due: [3.0](https://api.github.com/repos/yegor256/netbout/milestones/1). Please, do something about it.

**Comment by karato** [comment]
@dmzaytsev please, pay attention to this:

 * some tasks in the project are "stuck": #602 (54 days), #641 (48 days), #692 (31 days), #698 (29 days), #634 (29 days), #626 (29 days), #704 (26 days), #701 (26 days); please do something about it ASAP. You basically have three options: 1) mark some of them as "postponed" ([par.26](http://www.teamed.io/policy.html#26)), 2) re-assign them ([par.29](http://www.teamed.io/policy.html#29)), or 3) try to help their performers to resolve them.

 * some issues were reported some time ago, but still not dispatched: #656, #644, #622. Please, do something about them. Either register as bugs or ask their reporters to close them.

 * some milestones are due: [3.0](https://api.github.com/repos/yegor256/netbout/milestones/1). Please, either close them or update due dates.

**Comment by dmzaytsev** [comment]
@karato you can see status for the "stucked" task below
#602 not assigned yet
#641 seems the developer is offline. He is not respond. The task will be reassigned in two days.
#692 assigned two days ago
#698 postponed
#634 not assigned yet
#626 not assigned yet 
#704 assigned two days ago
#701 not assigned yet 

@yegor256 as far as I understand, if a task not assigned I can do nothing with such tasks. 
The project manager should dispatch the task. I understand architect should keep an eye on the "stuck" tasks, but I think it would better to break the report like this:

* not assigned tasks
* assigned less then 5 days ago
* assigned more then 5 days ago (no comments, no PR created, etc..). in my opinion it is a really stucked tasks. 
* the open issues but not marked as bug 

@yegor256 what do you think? make sense?

**Comment by yegor256** [comment]
@dmzaytsev make sense, but I think we shouldn't do this. From a project management perspective, a stuck task is a stuck task. Why it's stuck - doesn't really matter. It does matter for you, the architect. But from a project management point of view they are all the same - they have no progress for many days. Keep in mind that if/when a task is not assigned to anyone it means that there is not enough people in the project or something else is wrong with the "team". You should raise this concern right in the ticket and ask project manager to do something about it. See my point?

**Comment by dmzaytsev** [comment]
@yegor256 the projects managers respond to the template requests only. we need to know some key words for communications with them.. sometimes it seems to me they are not humans. the automation tools replaced the humans...
why they ask me to do something with the stuck tasks ASAP if tasks are still be not assigned for weeks even urgent tasks? There is not enough people in the project, I guess. But the managers has no information about it ? 
You proposed raise this concern right in the tickets. Ok I will try

**Comment by dmzaytsev** [comment]
@karato many tasks are still be not assigned even urgent tasks.
please do something about it

**Comment by karato** [comment]
@dmzaytsev There are a few problems in the project now:

 * some tasks are stuck: #602 (57 days), #641 (51 days), #692 (34 days), #634 (32 days), #626 (32 days), #704 (29 days), #701 (29 days), #703 (28 days); please resolve this. You have three options: 1) mark some of them as "postponed", 2) re-assign them, or 3) help performers to resolve them.

 * some issues were reported some time ago, but still not dispatched: #656, #644, #622. Please, do something about them. Either register as bugs or ask their reporters to close them.

 * some milestones are already due: [3.0](https://api.github.com/repos/yegor256/netbout/milestones/1). Please, do something about it.

**Comment by karato** [comment]
@dmzaytsev There are a few concerns:

 * some tasks are stuck: #602 (60 days), #692 (37 days), #634 (35 days), #626 (35 days), #703 (31 days), #713 (26 days), #714 (26 days), #697 (26 days); please resolve this. You have three options: 1) mark some of them as "postponed", 2) re-assign them, or 3) help performers to resolve them.

 * some issues were reported some time ago, but still not dispatched: #656, #644, #622. Please, do something about them. Either register as bugs or ask their reporters to close them.

 * some milestones are due: [3.0](https://api.github.com/repos/yegor256/netbout/milestones/1). Please, either close them or update due dates.

**Comment by karato** [comment]
@dmzaytsev There are a few problems in the project now:

 * some tasks are stuck: #602 (63 days), #692 (40 days), #626 (38 days), #703 (34 days), #713 (29 days), #714 (29 days), #715 (28 days), #717 (27 days); please resolve this. You have three options: 1) mark some of them as "postponed", 2) re-assign them, or 3) help performers to resolve them.

 * these issues are still not dispatched anyhow: #742, #656, #644, #622. Either register as bugs or ask their reporters to close.

 * there are some milestones with missed dates: [3.0](https://api.github.com/repos/yegor256/netbout/milestones/1). Please, do something about it.

**Comment by yegor256** [comment]
@dmzaytsev when are you planning to release this version to production? There are over 100 commits already in it.. maybe it's time to bring it live?

**Comment by dmzaytsev** [comment]
@yegor256 if you [approved](https://github.com/yegor256/netbout/issues/661#issuecomment-110810970) this  version we can release. It have some unresolved puzzles/bugs, but nothing to break the functionality.

>@dmzaytsev since we have a snapshot deployment, let's agree that you don't release to production until I approve what I see in the snapshot. OK?

**Comment by yegor256** [comment]
@dmzaytsev looks good for a release, let's do it

**Comment by dmzaytsev** [comment]
@rultor release, tag is `2.16`

**Comment by dmzaytsev** [comment]
@yegor256 please tag `2.16.1` as stable

**Comment by karato** [comment]
@dmzaytsev please, pay attention to this:

 * some tasks are stuck: #692 (59 days), #713 (49 days), #714 (48 days), #715 (48 days), #717 (47 days), #721 (45 days), #723 (45 days), #722 (44 days); please do something about it. There are basically three options: 1) mark some of them as "postponed" (see [par.26](http://www.teamed.io/policy.html#26)), 2) re-assign them (see [par.29](http://www.teamed.io/policy.html#29)), or 3) try to help their performers to resolve them.

 * these issues are still not dispatched anyhow: #656, #644, #622. Either register as bugs or ask their reporters to close.

 * there are some milestones with missed dates: [3.0](https://api.github.com/repos/yegor256/netbout/milestones/1), [3.1](https://api.github.com/repos/yegor256/netbout/milestones/2). Please, do something about it.... -100 to your rating, your total score is [+1792](http://www.netbout.com/b/35166?open=rating)

**Comment by dmzaytsev** [comment]
@yegor256 what is it? new rules??
> . -100 to your rating

I just checked [the policy](http://www.teamed.io/policy.html) and found nothing about it.
I believe the rules should by updated before applying. If rules changed the team should by notified before applying.

please restore my rating

**Comment by dmzaytsev** [comment]
@karato please set due date Oct 1 to milestone 3.0
and Dec 1 to milestone 3.1

**Comment by karato** [comment]
> @karato please set due date Oct 1 to milestone 3.0
> and Dec 1 to milestone 3.1

@dmzaytsev I attached this issue to milestone `3.1`

**Comment by yegor256** [comment]
@dmzaytsev I changed the dates of milestones and reverted the points. The rule in the policy is here: http://www.teamed.io/policy.html#33 It was introduced recently, I didn't announce it yet. But now you know, be careful

**Comment by dmzaytsev** [comment]
@yegor256 what is your decision? 
> @yegor256 please tag 2.16.1 as stable

**Comment by yegor256** [comment]
@karato please mark release 2.16.1 as final

**Comment by dmzaytsev** [comment]
@yegor256 I hope stucked tasks won't be fined... you know we haven't developers, tasks not dispatched a long time
should I ask the PM mark all tasks as postponed?

**Comment by yegor256** [comment]
@dmzaytsev I think I have to add more developers to the project. I'll try to do it ASAP

**Comment by dmzaytsev** [comment]
@yegor256 anyway even if you added developers, these tasks will be closed at least in 10 days.
if stuck tasks will be assigned...

**Comment by dmzaytsev** [comment]
@yegor256 I have to tell [it's](http://www.teamed.io/policy.html#33) unrealizable rule. Every 3-5 days the architect (me or someone else) will lose points on this project. We have some "stuck" tasks, some of them is postponed. First, even if some developers will be assigned, anyway it may take several weeks, but there will be new stuck tasks. It is an endless process...
Second, if the architect will put a postponed task to the normal state, he/she will be fined immediately because the task became "stuck". There is only one way to follow the rule put all task to the postponed state and never touch them. But this is not acceptable way, isn't it?

**Comment by karato** [comment]
@dmzaytsev please, pay attention to this:

 * some tasks are stuck: #692 (64 days), #713 (53 days), #714 (53 days), #715 (52 days), #717 (51 days), #721 (50 days), #723 (49 days), #726 (47 days); please do something about it. There are basically three options: 1) mark some of them as "postponed" (see [par.26](http://www.teamed.io/policy.html#26)), 2) re-assign them (see [par.29](http://www.teamed.io/policy.html#29)), or 3) try to help their performers to resolve them.

 * some issues were reported some time ago, but still not dispatched: #757, #622. Please, do something about them. Either register as bugs or ask their reporters to close them.

-25 added to your rating, at the moment it is: [+1596](http://www.netbout.com/b/35166?open=rating)

**Comment by dmzaytsev** [comment]
@yegor256 I guess I was fined for this, right?

> some issues were reported some time ago, but still not dispatched: #757, #622. Please, do something about them. Either register as bugs or ask their reporters to close them.

if so, that's unfairly 
I asked tag #757 as a bug [5 days ago](https://github.com/yegor256/netbout/issues/757#issuecomment-129424436)

same about #622, I postponed it twice [Jul 11] (https://github.com/yegor256/netbout/issues/622#issuecomment-120626856) and [Aug 12](https://github.com/yegor256/netbout/issues/622#issuecomment-130356311)

**Comment by yegor256** [comment]
@dmzaytsev our fault indeed. I removed the deduction from your rating.

**Comment by dmzaytsev** [comment]
@karato please mark release 2.16.1 as final

> yegor256 commented 5 days ago
> @karato please mark release 2.16.1 as final

**Comment by yegor256** [comment]
@karato release 2.16.1 is final

**Comment by karato** [comment]
> @karato please mark release 2.16.1 as final
> 
> > yegor256 commented 5 days ago
> > @karato please mark release 2.16.1 as final
> 
> 

@dmzaytsev OK, will do now

**Comment by karato** [comment]
> @karato please mark release 2.16.1 as final
> 
> > yegor256 commented 5 days ago
> > @karato please mark release 2.16.1 as final
> 
> 

@dmzaytsev OK, will do now

**Comment by karato** [comment]
> @karato please mark release 2.16.1 as final
> 
> > yegor256 commented 5 days ago
> > @karato please mark release 2.16.1 as final
> 
> 

@dmzaytsev OK, will do now

**Comment by karato** [comment]
> @karato release 2.16.1 is final

@yegor256 done, release 2.16.1 is final, the "draft" flag is removed

**Comment by karato** [comment]
@dmzaytsev I am closing it, thanks all

**Comment by karato** [comment]
@dmzaytsev Much obliged! I have added **30 mins** to your account in payment "63468871", 1674 hours and 54 mins spent... +30 added to your rating, current score is: [+1564](http://www.netbout.com/b/35166?open=rating)

**Comment by dmzaytsev** [comment]
@yegor256 is it final value? only 30 min?

**Comment by yegor256** [comment]
@dmzaytsev the bonus here is less than a few minutes, the numbers are t=1656, c=107, h=5732, diff=2.15.1..master, this is the calculator: t=1656, c=107, h=5732, diff=2.15.1..master
